pennylane/templates/subroutines/prepselprep.py

```diff
@@ -23,22 +23,16 @@
 
 def _get_new_terms(lcu):
     """Compute a new sum of unitaries with positive coefficients"""
-
-    new_coeffs = []
+    coeffs, ops = lcu.terms()
     new_ops = []
 
-    for coeff, op in zip(*lcu.terms()):
-
+    for coeff, op in zip(coeffs, ops):
```

Comment:
Maybe we can compute `angles = qml.math.angle(coeffs)` outside the loop and then zip over `angles` and `ops`.